lib/segment/src/index/field_index/geo_index/mod.rs

```diff
@@ -103,8 +104,9 @@ impl GeoMapIndex {
         format!("{field}_geo")
     }
 
-    fn encode_db_key(value: &str, idx: PointOffsetType) -> String {
-        format!("{value}/{idx}")
+    fn encode_db_key(value: GeoHash, idx: PointOffsetType) -> String {
```

Comment:
Nit: this method could return `-> SmolStr` to avoid allocation.

Reply:
The `format!` could be replaced with `write!`.

```rust
use std::fmt::Write as _;
let mut b = smol_str::SmolStrBuilder::new();
write!(b, "{value}/{idx}").unwrap();
b.finish()
```

Using unwrap here is fine (aka no worse than using `format!`), see
- [`format!` doc on when it panics](https://doc.rust-lang.org/std/macro.format.html#panics).
- [`impl fmt::Write for SmolStrBuilder`](https://docs.rs/smol_str/0.3.1/src/smol_str/lib.rs.html#801-807) never returns an `Error`.

Reply:
Ah, even better, there is [`format_smolstr!`](https://docs.rs/smol_str/latest/smol_str/macro.format_smolstr.html) macro.

Reply:
Great! Thanks for googling it for me, fixed!